README.md

```diff
@@ -0,0 +1,275 @@
+```
+
+Then use it in your Terraform configuration:
+```hcl
+module "docker_build" {
+  source = "github.com/METR/terraform-docker-build"
+
+  ecr_repo     = "my-app"
+  source_path  = "${path.module}/app"
+  source_files = ["**/*"]
+  
+  builder      = "yourorg/yourbuilder"
+  platform     = "linux/amd64,linux/arm64"
+}
+```
+
+## Migration Guide
+
+### Migrating from terraform-aws-modules/lambda/aws//modules/docker-build
+
+This module serves as a drop-in replacement for the AWS Lambda docker-build module. Update your module source:
+
+```hcl
+# Before
+module "docker_build" {
+  source  = "terraform-aws-modules/lambda/aws//modules/docker-build"
+  version = "~> 7.21"
+  
+  providers = {
+    docker = docker
+  }
+  
+  # Your existing configuration
+  source_path = "./src"
+  # ... other variables
+}
+
+# After
+module "docker_build" {
+  source = "github.com/METR/terraform-docker-build"
+  
+  # Same configuration, but remove the providers block
+  ecr_repo     = "your-repo-name"  # This replaces any ECR setup
+  source_path  = "./src"
+  source_files = ["**/*"]          # Add this for file tracking
+  # ... other compatible variables
+}
+```
+
+### Migrating from kreuzwerker/docker Provider
+
+If you're currently using the [kreuzwerker/docker](https://github.com/kreuzwerker/terraform-provider-docker) provider with `docker_registry_image` resources, follow these steps:
+
+**Note**: If you're running in an environment without Docker daemon access (like Spacelift's public runners), you'll need to add `disable_docker_daemon_check = true` to your docker provider configuration during the migration process.
+
+1. **Update your configuration** to use this module:
+   ```hcl
+   # Before (using docker provider)
+   terraform {
+     required_providers {
+       docker = {
+         source  = "kreuzwerker/docker"
+         version = "~> 3.0"
+       }
+     }
+   }
+
+   provider "docker" {
+     registry_auth {
+       address  = data.aws_ecr_authorization_token.token.proxy_endpoint
+       username = data.aws_ecr_authorization_token.token.user_name
+       password = data.aws_ecr_authorization_token.token.password
+     }
+   }
+
+   resource "docker_image" "this" {
+     name = "my-app:latest"
+     build {
+       context = "./src"
+     }
+   }
+
+   resource "docker_registry_image" "this" {
+     name = docker_image.this.name
+     triggers = {
+       dir_sha1 = sha1(join("", [for f in fileset("./src", "**") : filesha1("./src/${f}")]))
+     }
+   }
+
+   # After (using this module)
+   module "docker_build" {
+     source = "github.com/METR/terraform-docker-build"
+     
+     ecr_repo     = "my-app"
+     source_path  = "./src"
+     source_files = ["**/*"]
+   }
+   ```
+
+2. **Authenticate with ECR** (replace region and account ID with your values):
+   ```bash
+   aws ecr get-login-password --region us-west-1 | docker login --username AWS --password-stdin YOUR_ACCOUNT_ID.dkr.ecr.us-west-1.amazonaws.com
+   ```
```

Comment:
Very minor: to be platform-agnostic/up to date, I'd be tempted to link to AWS's documentation here https://docs.aws.amazon.com/AmazonECR/latest/userguide/registry_auth.html